### PR TITLE
feature/issue-408/entity-panel

### PR DESCRIPTION
**Pull request by rostyslavvnahornyi**

### CODE REVIEW

**Comment by kfarr** [comment]
* addlayerbutton needs focus state (implemented correctly on addlayerpanel closeButton)
* How to populate menu - alex to work on that

**Comment by Algorush** [comment]
I added new parameter mixinGroups to options array. For specifying one or few mixin group names to show if option is selected:
```
  const options = [
    {
      value: 'Layers: Streets & Intersections',
      label: 'Layers: Streets & Intersections',
      onClick: () => console.log('Layers: Streets & Intersections')
    },
    {
      value: 'Models: Personal Vehicles',
      label: 'Models: Personal Vehicles',
      mixinGroups: ['vehicles', 'vehicles-rigged']
    },
    {
      value: 'Models: Transit Vehicles',
      label: 'Models: Transit Vehicles',
      mixinGroups: ['vehicles-transit']
    },
```
I sugest to store cardsData with objects data (mixin id, img, name, description) in separate json file. And options object with mixin group and Layers data in separate json file too for convenient. Where is better to store these files?
One question about options array. Now there are many more categories of mixins than group names in options. 
I have done it now so that all mixins are displayed, without dividing them into subgroups in selectedOption.
I see few variants to do here:
- display mixins by groups in the panel for convenience. Based on data from the mixinGroups parameter. 
Then we need to add styles to divide the list of objects to be added into groups. 
- Or divide Options in mixin subgroups and display a option list with mixin subgroups.
- use sumething like hashtags for selecting mixin subgroups

File: src/components/components/AddLayerPanel/AddLayerPanel.component.jsx
Comment (Algorush) [inline src/components/components/AddLayerPanel/AddLayerPanel.component.jsx:1]: @rostyslavvnahornyi, hello, could you help why I cant add file with cardsData like this?

**Comment by Algorush** [comment]
I added the ability to load layers by clicking on cards. And also svg-extruder element. All this is for testing. For now, all this is in one place - 'Layers: Streets & Intersections', it could be distributed among other groups later. All this works, but for example, when deleting a mapbox element from the scene, an error occurs in 3dstreet-editor, and also when deleting a new street, for some reason dummy elements are added to the structure of the existing street, after opening/closing the editor everything is fine. I will create isseus in Editor for it

**Comment by kfarr** [comment]
remaining before this can be merged:
- [ ] - test with saving
- [ ] - review logic for where to place these new objects -- as children or elsewhere?
- [ ] - review options for where to store json catalog of objects and how to generate catalog
- [ ] - fix scrolling behavior and other style issues (menu)

**Comment by Algorush** [comment]
Now adding and previewing objects correspond to the elevation level of the segment for which they are added. And also in this PR, I made that objects are always added as direct descendants of the parent segment of the currently selected entity in the scene. In another [branch](https://github.com/3DStreet/3dstreet-editor/pull/421), objects are always added to a special entity with class .custom-group

**Comment by kfarr** [comment]
Checklist to complete v1:

- [ ] Choose which branch we are using for this -- there are 2?
- [ ] Errors while hovering for some objects - triage first
- [ ] Panel should be hidden behind a isBeta claim flag
- [ ] Review -- are these the layers that we need? Any missing? Any shouldn't be there?

LATER:
- [ ] LATER addlayerbutton needs focus state (implemented correctly on addlayerpanel closeButton)
- [ ] LATER Long-term should use a catalog.json format that is shared between core and editor. (Maybe even bring core into editor)
- [ ] LATER Long-term issue -- Hover preview – Is it always creating new entities or is it changing the value of 1 entity? Is it properly disposing of the gltf preview?
- [ ] LATER -- how to create input mechanism for user when placing an entity that has required properties (such as a src url when placing an image)

**Comment by kfarr** [comment]
see #421

File: src/components/components/AddLayerPanel/AddLayerPanel.component.jsx [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+    activeCamera.getWorldDirection(direction);
+    cameraFrontVec.copy(activeCamera.position);
+    cameraFrontVec.add(direction.multiplyScalar(5));
+
+    // place preview entity in front of camera in 5 meters
+    preEntity.setAttribute('position', cameraFrontVec);
+  };
+
+  const getParentSegmentEl = (element) => {
+    if (element.className.includes('segment-parent')) {
```
Comment (vincentfretin) [inline src/components/components/AddLayerPanel/AddLayerPanel.component.jsx:240]: It depends of what you want to do here, I don't see segment-parent used anywhere so I'm not sure what is the intended behavior of this code.
If you want to test if the element has the class "segment-parent", it's preferable to do
`element.classList.contains('segment-parent'))`
The line `element.className.includes('segment-parent')` will match any class with segment-parent in it, like `prefix-segment-parent-suffix`.

https://developer.mozilla.org/en-US/docs/Web/API/Element/classList

File: src/components/components/AddLayerPanel/AddLayerPanel.component.jsx [hunk 2]
```diff
@@ -0,0 +1,356 @@
...
+  AFRAME.scenes[0].appendChild(preEntity);
+
+  const previewEntity = () => {
+    if (selectedOption == 'Models: Buildings') {
+      preEntity.setAttribute('scale', { x: 0.1, y: 0.1, z: 0.1 });
+    } else {
+      preEntity.setAttribute('scale', { x: 0.5, y: 0.5, z: 0.5 });
+    }
+    /* place object in front of active camera */
+    const direction = new THREE.Vector3();
```
Comment (vincentfretin) [inline src/components/components/AddLayerPanel/AddLayerPanel.component.jsx:228]: You can move direction var out of the function so it's not created every time.

**Review by vincentfretin — COMMENTED** [review]
I see this PR is now closed so not sure if my comment is useful, but here it is.

File: src/components/components/AddLayerPanel/AddLayerPanel.component.jsx (hunk 1 above)
Comment (vincentfretin) [reply src/components/components/AddLayerPanel/AddLayerPanel.component.jsx:240]: This comment also applies to #421

File: src/components/components/AddLayerPanel/AddLayerPanel.component.jsx (hunk 2 above)
Comment (vincentfretin) [reply src/components/components/AddLayerPanel/AddLayerPanel.component.jsx:228]: This comment also applies to #421

**Pull request closed without merging**